src/Storage/Factory/AbstractFactory.php

```diff
@@ -37,4 +40,26 @@ protected function usageResolver(): \Closure
             throw new \RuntimeException(\sprintf('Unsupported storage usage value %s', $value));
         };
     }
+
+    protected function hotSynchronizeLimitResolver(): \Closure
+    {
+        return function (Options $options, string $value): int {
+            $matches = [];
+            \preg_match('/^\s*(?P<number>\d+)\s*(?:(?P<prefix>[kmgt]?)b?)?\s*$/i', $value, $matches);
+
+            $limit = \intval($matches['number'] ?? 0);
+            $prefix = \strtolower(\strval($matches['prefix'] ?? ''));
+            switch ($prefix) {
+                case 't': $limit *= 1024;
+                // no break
+                case 'g': $limit *= 1024;
+                // no break
+                case 'm': $limit *= 1024;
+                // no break
+                case 'k': $limit *= 1024;
+            }
+
+            return $limit;
```

Comment:
they all do the same? maybe:

```suggestion
            return \in_array($prefix, ['t', 'g', 'm', 'k'], true) ? ($limit * 1024) : $limit;
```

Reply:
Yes but no. You miss the magic part (`// no break`)- ;-):

- 1k should give 1024 (1*1024)
- 1m should give 1024*1024
- 1g should give 1024*1024*1024
- ....

Reply:
ooh yes